EESSI-install-software.sh

```diff
@@ -278,6 +262,52 @@ if command_exists "nvidia-smi"; then
     ${EESSI_PREFIX}/scripts/gpu_support/nvidia/link_nvidia_host_libraries.sh
 fi
 
+
+echo ">> Configuring EasyBuild..."
+
+# Make sure that we use the EESSI_CVMFS_INSTALL
+# Since the path is set when loading EESSI-extend, we reload it to make sure it works - even if it is already loaded
+# Note we need to do this after running install_cuda_and_libraries, since that does installations in the EESSI_SITE_INSTALL 
+unset EESSI_USER_INSTALL
+unset EESSI_PROJECT_INSTALL
+unset EESSI_SITE_INSTALL
+export EESSI_CVMFS_INSTALL=1
+module unload EESSI-extend
+module load EESSI-extend/${EESSI_VERSION}-easybuild
+
+if [ ! -z "${shared_fs_path}" ]; then
+    shared_eb_sourcepath=${shared_fs_path}/easybuild/sources
+    echo ">> Using ${shared_eb_sourcepath} as shared EasyBuild source path"
+    export EASYBUILD_SOURCEPATH=${shared_eb_sourcepath}:${EASYBUILD_SOURCEPATH}
+fi
+
+# if an accelerator target is specified, we need to make sure that the CPU-only modules are also still available
+if [ ! -z ${EESSI_ACCELERATOR_TARGET} ]; then
+    CPU_ONLY_MODULES_PATH=$(echo $EASYBUILD_INSTALLPATH | sed "s@/accel/${EESSI_ACCELERATOR_TARGET}@@g")/modules/all
+    if [ -d ${CPU_ONLY_MODULES_PATH} ]; then
+        module use ${CPU_ONLY_MODULES_PATH}
+    else
+        fatal_error "Derived path to CPU-only modules does not exist: ${CPU_ONLY_MODULES_PATH}"
+    fi
+fi
+
+# If in dev.eessi.io, allow building on top of softw
+if [[ "${EESSI_CVMFS_REPO}" == /cvmfs/dev.eessi.io ]]; then
+    module use /cvmfs/software.eessi.io/versions/$EESSI_VERSION/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR_OVERRIDE}/modules/all
```

Comment:
Would be good to make a comment that explains the expected precedence in `MODULEPATH`:
```
/path/to/dev.eessi.io/modules:/path/to/accel/modules:/path/to/software.eessi.io/modules
```

Reply:
I'm not sure you have caught the case though that would define this:
* I want to install in dev.eessi.io
* I want to install for an accelerator (and require accelerator dependencies)

Reply:
I wonder is this what we want to support:
```
/path/to/dev.eessi.io/accel/modules:/path/to/dev.eessi.io/modules:/path/to/software.eessi.io/accel/modules:/path/to/software.eessi.io/modules
```

Reply:
While my pr _moved_ this section of code, I'm pretty sure it was already there, wasn't it? I'm on my phone currently, so can't easily check

Reply:
I'd be in favour of doing that in a separate PR, if @ocaisa agrees. This PR currently is mostly a reshuffle: some parts that can only be done after EESSI is initialized got moved down until after the EESSI module is loaded. While the pr replaced sourcing the various init scripts with loading the module, it _should_ not modify behaviour in any way. I think it's more clear of we do a big fix like that in a separate PR

Reply:
I've opened https://github.com/EESSI/software-layer/issues/799 to keep track of this